File: src/levels/L06-decorator/ts/subclasses.ts
class Latte implements Drink {
  cost() { return 45 }
  description() { return 'Latte' }
}

// Una subclase por cada combinación de extras.
// region: LatteConAvena
class LatteConAvena extends Latte {
  cost() { return super.cost() + 8 }
  description() { return 'Latte con avena' }
}
// endregion

// region: LatteConShot
class LatteConShot extends Latte {
  cost() { return super.cost() + 10 }
  description() { return 'Latte con shot' }
}
// endregion

// region: LatteConAvenaYShot
class LatteConAvenaYShot extends Latte {
  cost() { return super.cost() + 18 }
  description() { return 'Latte con avena y shot' }
}
// endregion

File: src/levels/L06-decorator/subclasses.rb
class Latte
  def cost = 45
  def description = "Latte"
end

# Una subclase por cada combinación de extras.
# region: LatteConAvena
class LatteConAvena < Latte
  def cost = super + 8
  def description = "Latte con avena"
end
# endregion

# region: LatteConShot
class LatteConShot < Latte
  def cost = super + 10
  def description = "Latte con shot"
end
# endregion

# region: LatteConAvenaYShot
class LatteConAvenaYShot < Latte
  def cost = super + 18
  def description = "Latte con avena y shot"
end
# endregion
